packages/terser/src/module.ts

```diff
@@ -0,0 +1,67 @@
+import { NormalizedOutputOptions, RenderedChunk } from 'rollup';
+import { hasOwnProperty, isObject, merge } from 'smob';
+
+import { Options } from './type';
+import { callWorker } from './worker';
+
+export default function terser(options: Options = {}) {
+  return {
+    name: 'terser',
+
+    async renderChunk(code: string, chunk: RenderedChunk, outputOptions: NormalizedOutputOptions) {
+      const defaultOptions: Options = {
+        sourceMap: outputOptions.sourcemap === true || typeof outputOptions.sourcemap === 'string'
+      };
+
+      if (outputOptions.format === 'es') {
+        defaultOptions.module = true;
+      }
+
+      if (outputOptions.format === 'cjs') {
+        defaultOptions.toplevel = true;
+      }
+
+      try {
+        const { code: result, nameCache } = await callWorker(__filename, {
```

Comment:
Is this going to create a new Worker for each call to `renderChunk`?

From the [Worker threads](https://nodejs.org/api/worker_threads.html) doc:

> In practice, use a pool of Workers for these kinds of tasks. Otherwise, the overhead of creating Workers would likely exceed their benefit.

The doc also links to an example pool implementation: https://nodejs.org/api/async_context.html#using-asyncresource-for-a-worker-thread-pool

Reply:
Yes, at the moment that is the case.
The idea was to let the operating system decide when enough resources are available to run unprocessed worker threads.
But in general it might be better to have more control over it, to better control the cpu usage.

Reply:
@dasa  it is now possible to cotnrol how many workers can run at the same time. By default it is equal to the amount of cpu cores.

Reply:
It doesn't appear to me that the workers are being reused. If that's the case, it seems this would still be true, "the overhead of creating Workers would likely exceed their benefit."